Source/Core/Common/BitUtils.h

```diff
@@ -240,6 +241,50 @@ inline auto BitCastPtr(PtrType* ptr) noexcept -> BitCastPtrType<T, PtrType>
   return BitCastPtrType<T, PtrType>{ptr};
 }
 
+// Similar to BitCastPtr, but specifically for aliasing structs to arrays.
+template <typename ArrayType, typename T,
+          typename Container =
+              std::array<ArrayType, (sizeof(T) + sizeof(ArrayType) - 1) / sizeof(ArrayType)>>
```

Comment:
Is there really a case for supporting `sizeof(T) % sizeof(ArrayType) != 0` ? seems like it could cause surprising bugs in the future.

Reply:
Yeah I don't think there's any reasonable circumstances under which you'd use it like that. Should we just enforce alignment with `ArrayType` using an assert?